### PR TITLE
Sorted/Ordered Bookmarks

### LINKED ISSUE #822: Weired Bookmarks behaviour when having more bookmarks
- Add more than 3 bookmarks (probably issue and repo) 
- constantly tap between search and bookmark tab 

Notice that bookmarks order is changing and on click it's going to wrong bookmark then what it displays.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

### PR DESCRIPTION
**Pull request by rizwankce**
Closes #822 

Learnt some nifty details. Oversight from my side. Hope I'm not alone

### CODE REVIEW

File: Classes/Bookmark/BookmarkStore.swift [hunk 1]
```diff
@@ -75,7 +75,7 @@ final class BookmarksStore {
 
     var bookmarks: [BookmarkModel] {
         refresh()
-        return Array(_bookmarks)
+        return Array(_bookmarks.sorted(by: { "\($0.owner)\($0.name)\($0.number)" > "\($1.owner)\($1.name)\($1.number)"}))
```
Comment (BasThomas) [inline Classes/Bookmark/BookmarkStore.swift]: What about making `BookMarkModel` conform to `Comparable` and using `>`?

Comment (Sherlouk) [reply Classes/Bookmark/BookmarkStore.swift]: Also is sorting the array _every_ time we fetch this value the best idea? I wonder if there's a better place we could do this?

**Comment by rnystrom** [comment]
This will sort every single time `bookmarks` is called. Maybe instead it should sort when you mutate the store?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by rizwankce** [comment]
Guess we will end up in the same situation when we get the bookmarks.  I'll give it a try though 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by rizwankce** [comment]
After a bit of rethinking, We actually need a collection which can avoid duplicates and maintain the order. So end up giving away `Set<BookmarkModel>` and added `NSMutableOrderedSet`.

So by this way, we can avoid adding `sorted` method and some overkills comes along with it.

**Note** 
If this PR accepted. Possibly all your previous bookmarks will be lost.

File: Classes/Bookmark/BookmarkStore.swift [hunk 2]
```diff
@@ -36,7 +36,7 @@ final class BookmarksStore {
     }
 
     private var token: String
-    private var _bookmarks: Set<BookmarkModel> = []
+    private var _bookmarks: NSMutableOrderedSet = []
```
Comment (rnystrom) [inline Classes/Bookmark/BookmarkStore.swift]: wo TIL this syntax works

File: Classes/Bookmark/BookmarkStore.swift [hunk 3]
```diff
@@ -69,19 +69,23 @@ final class BookmarksStore {
...
     }
 
     var bookmarks: [BookmarkModel] {
         refresh()
-        return Array(_bookmarks)
+        guard let bookmarks = _bookmarks.array as? [BookmarkModel] else {
+            return []
+        }
+
+        return bookmarks
```
Comment (rnystrom) [inline Classes/Bookmark/BookmarkStore.swift]: great!

**Pull request merged**